pkg/util/camel/camel_runtime_catalog.go

```diff
@@ -183,7 +183,17 @@ func (c *RuntimeCatalog) VisitSchemes(visitor func(string, v1.CamelScheme) bool)
 
 // DecodeComponent parses the given URI and return a camel artifact and a scheme.
 func (c *RuntimeCatalog) DecodeComponent(uri string) (*v1.CamelArtifact, *v1.CamelScheme) {
-	uriSplit := strings.SplitN(uri, ":", 2)
+
+	var uriSplit []string
+
+	//Decode URI using formats http://my-site/test?param=value or log:info
+	if strings.Contains(uri, "?") {
```

Comment:
The function should be able to detect the component when there is no context or path parameter, as in 
```
uri: direct
```
As stated in the bug report.

Add a test like this, it will fail:
```
{desc: "Basic static dependency witch path and name", uri: "direct", expectedID: "direct"},
```